### PR TITLE
`Development`: Update file upload exercise documentation

### PR DESCRIPTION
**Pull request by julian-christl**
#### General

- [X] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [X] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

Part of your documentation improvement marathon.

### Description

There was not much to change as it's a pretty straightforward feature. I decided to adapt the phrasing in a few parts and, other than that, updated the images.

Additionally, I moved some images that both the file upload and the programming section were using to a general folder. Others may use that folder too. We can also move it up the hierarchy if someone deems this necessary.

Documentation: https://artemis-platform--5428.org.readthedocs.build/en/5428/user/exercises/file-upload/

#### Code Review
- [ ] Review 1
- [ ] Review 2

### CODE REVIEW

File: docs/user/exercises/file-upload.rst [hunk 1]
```diff
@@ -51,12 +51,12 @@ Student Submission
 - When the exercise is released the students can upload the requested file.
```
Comment (babyygemperor) [inline docs/user/exercises/file-upload.rst:51]: is released, the students can (missing comma)

File: docs/user/exercises/file-upload.rst [hunk 2]
```diff
@@ -65,10 +65,13 @@ Assessment
 - When the due date is over you can assess the submissions. From the assessment dashboard go to exercise assessment dashboard of the file upload exercise.
```
Comment (babyygemperor) [inline docs/user/exercises/file-upload.rst:65]: assessment dashboard, go to exercise (missing comma)

File: docs/user/exercises/file-upload.rst [hunk 3]
```diff
@@ -65,10 +65,13 @@ Assessment
...
 - There you can assess the submitted student submissions, by first downloading the file, and then creating feedback with points.
```
Comment (babyygemperor) [inline docs/user/exercises/file-upload.rst:66]: Perhaps "adding feedback with points"?

File: docs/user/exercises/file-upload.rst [hunk 4]
```diff
@@ -23,7 +23,7 @@ The following sections describe the supported features and the process of creati
 - Open |course-management|
```
Comment (babyygemperor) [inline docs/user/exercises/file-upload.rst:23]: Missing full stop at the end

File: docs/user/exercises/file-upload.rst [hunk 5]
```diff
@@ -23,7 +23,7 @@ The following sections describe the supported features and the process of creati
...
 - Navigate into **Exercises** of your preferred course
```
Comment (babyygemperor) [inline docs/user/exercises/file-upload.rst:24]: Missing full stop at the end

File: docs/user/exercises/file-upload.rst [hunk 6]
```diff
@@ -23,7 +23,7 @@ The following sections describe the supported features and the process of creati
...
 
-    .. figure:: programming/course-management-course-dashboard.png
+    .. figure:: general/course-management-course-dashboard-exercises.png
               :align: center
 
 - Click on **Create new file upload exercise**
```
Comment (babyygemperor) [inline docs/user/exercises/file-upload.rst:29]: Missing full stop at the end

File: docs/user/exercises/file-upload.rst [hunk 7]
```diff
@@ -32,17 +32,17 @@ The following sections describe the supported features and the process of creati
               :align: center
 
 - Fill out all mandatory values and click on |save|
```
Comment (babyygemperor) [inline docs/user/exercises/file-upload.rst:34]: Missing full stop at the end

**Review by babyygemperor — CHANGES_REQUESTED** [review]
Although I do realise that most of my requested changes are not from this PR, but it would be great if we could cover them in this PR.

File: docs/user/exercises/file-upload.rst (hunk 2 above)
Comment (jschmautzer) [inline docs/user/exercises/file-upload.rst:65]:
```suggestion
- When the due date is over, you can assess the submissions. From the assessment dashboard go to the exercise assessment dashboard of the file upload exercise.
```

**Review by jschmautzer — COMMENTED** [review]
only some small things :) 
Rest looks good.

**Review by nadinesamyy — COMMENTED** [review]
Maybe in the sentence "Fill out all mandatory values and click on", the word "values" could be replaced by "fields", other than that it looks good

**Review by jbllhff — COMMENTED** [review]
Looks good after taking the previously mentioned comments into account.

**Review by joaquinmurcia — CHANGES_REQUESTED** [review]

**Review by MetinLamby — APPROVED** [review]
looks very good!

File: docs/user/exercises/file-upload.rst [hunk 8]
```diff
@@ -51,12 +51,12 @@ Student Submission
 - When the exercise is released the students can upload the requested file.
 - They can choose a file with |browse| and then submit the file with |submit|.
 
-    .. figure:: file-upload/file_upload_exercise_student_view.png
+    .. figure:: file-upload/file-upload-exercise-student-view.png
               :align: center
 
 - After a file was uploaded it can then be downloaded again with the link below.
```
Comment (alaraozdenler) [inline docs/user/exercises/file-upload.rst:57]: After a file "is" uploaded", it can be" downloaded again with the link below.

**Review by alaraozdenler — COMMENTED** [review]
Just one minor reformulation suggestion

File: docs/user/exercises/file-upload.rst [hunk 9]
```diff
@@ -11,9 +11,9 @@ Overview
 
 Conducting a file upload exercise consists of 3 steps:
 
-1. **Instructor prepares exercise:** Creates and configures the exercise in Artemis.
-2. **Student solves exercise:** Creates a submission file, and uploads it.
-3. **Instructor assesses submissions:** Review the handed in files and create manual results for the students.
+1. Setup of the exercie in Artemis
+2. Submission of a student
```
Comment (manuelmanso) [inline docs/user/exercises/file-upload.rst:15]:
```suggestion
2. Submission by a student
```
Small correction

**Review by manuelmanso — APPROVED** [review]
LGTM, just a small correction along with what others have already suggested

File: docs/user/exercises/file-upload.rst [hunk 10]
```diff
@@ -32,17 +32,17 @@ The following sections describe the supported features and the process of creati
...
 
-  Result: **File Upload Exercise**
+- Result: **File Upload Exercise**
 
     .. figure:: file-upload/course-dashboard-exercise-file-upload.png
               :align: center
 
-- Click the |edit| button of the file upload exercise and adapt the interactive problem statement. There you can also set release and due dates.
+- Click the |edit| button of the file upload exercise to update the configuration and assessment instructions.
 - You can get an overview of the exercise by clicking on the title.
```
Comment (nairaarnez) [inline docs/user/exercises/file-upload.rst:46]: I would change this to: instructors and tutors can get an

**Review by nairaarnez — APPROVED** [review]
Other than that it looks good!

**Review by SimonPaul02 — APPROVED** [review]
I agree with the suggestions from other reviewers. The rest looks good to me

**Review by yuling-sun — APPROVED** [review]
Looks good for me. I didn't find any additional mistakes aside from those already mentioned.

**Review by nairaarnez — APPROVED** [review]
Approved after feedback was incorporated!

**Review by jschmautzer — APPROVED** [review]
re-approve (looks good to me)

**Review by alaraozdenler — APPROVED** [review]
Documentation looks good now!

**Pull request merged**